lib/snapshot/reader.js

```diff
@@ -9,21 +10,44 @@ module.exports = {
   processFunctionsToInspect,
 };
 
-function getLatest() {
-  return getFunctionsToInspect();
+function getLatest(local) {
+  let rawSnapshot;
+  const projectId = 'change me!';
+  if (local) {
+    debug('loading local snapshot');
+    rawSnapshot = fromLocal();
+  } else {
+    debug('loading snapshot from upstream');
+    rawSnapshot = fromUpstream(projectId);
+  }
+
+  return processFunctionsToInspect(rawSnapshot);
+}
+
+async function fromUpstream(projectId) {
+  // TODO AMIR: refactor config
+  const SNAPSHOT_BASE_URL = 'https://homebase.snyk.io';
+  const SNAPSHOT_URL_PATH = `/api/v1/snapshot/${projectId}/js`;
+  const url = SNAPSHOT_BASE_URL + SNAPSHOT_URL_PATH;
+  try {
+    return await needle('get', url);
+  } catch (error) {
+    debug(`failed retrieving latest snapshot from ${url}: ${error}`);
```

Comment:
I don't think `debug` is the appropriate log level here. Nor do I think we should be reverting to an empty object. Something needs to do proper error handling, and revert to some sensible behaviour. Log and delete is not appropriate. I don't mind if it's inside or outside this function.

Reply:
regarding the log level - it's not a log level, it's the library we use to print anything right now, except for actual crashes of the agent. so it's really either `debug` or nothing.
regarding the logic - what would make it more sensible then?
throwing an error, catching it _somewhere_ and _then_ deciding that failure to retrieve newer snapshots equals keeping the existing snapshot?